server/api/itemRouter.js

```diff
@@ -0,0 +1,123 @@
+const Item = require('../db/models/item');
+const Category = require('../db/models/category');
+const express = require('express');
+const router = express.Router();
+
+/**
+ * Add an item
+ * 
+ * req.body: {sourceLink: String,
+ *            highlight: String,
+ *            userId: mongoose.ObjectId,
+ *            placesId: id(optional),
+ *            comment: String (optional),}
+ * 
+ * response: {success: boolean,
+ *            message: String}
+ */
+router.post('/addItem', (req, res, next) => {
+  const newItem = req.body;
```

Comment:
Same comment as above, can this statement error, and should it be handled?

Reply:
I think at worse req.body is undefined, which is why previously i didn't include it in try blocks. If you pass in an undefined argument to a mongoose call, and that throws an error, then technically the error is handled there.

Reply:
we could handle these by doing if (newCategory) then proceed with the rest. Though i'm not sure that's necessary. Either works probably?

Reply:
I don't know enough to say. The server will currently respond with a 500 error, but I think 400 would be more appropriate (malformed request)

Reply:
the error response would be from next(error), and the status code would depend on the error from the mongodb call's callback or whatever error the method catches. I don't think that is caused by req.body. 

If we want to change the error code in general, we can write a middleware error function to make it always give 400 as code

Reply:
I see, thank you!